src/lsdb/core/search/index_search.py

```diff
@@ -1,34 +1,49 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING
+from typing import Any
 
 import nested_pandas as npd
-from hats.catalog.index.index_catalog import IndexCatalog
+import numpy as np
+from hats.catalog.index.index_catalog import IndexCatalog as HCIndexCatalog
 
 from lsdb.core.search.abstract_search import AbstractSearch
-
-if TYPE_CHECKING:
-    from lsdb.types import HCCatalogTypeVar
+from lsdb.types import HCCatalogTypeVar
 
 
 class IndexSearch(AbstractSearch):
-    """Find rows by ids (or other value indexed by a catalog index).
+    """Find rows by column values using HATS index catalogs."""
 
-    Filters partitions in the catalog to those that could contain the ids requested.
-    Filters to points that have matching values in the id field.
+    values: dict[str, Any]
+    """Mapping of field name to the value we want to match it to"""
 
-    NB: This requires a previously-computed catalog index table.
-    """
+    index_catalogs: dict[str, HCIndexCatalog]
+    """Mapping of field name to respective index catalog"""
 
-    def __init__(self, ids, catalog_index: IndexCatalog, fine: bool = True):
-        super().__init__(fine)
-        self.ids = ids
-        self.catalog_index = catalog_index
+    def __init__(self, values: dict[str, Any], index_catalogs: dict[str, HCIndexCatalog]):
+        super().__init__(fine=True)
+        if values.keys() != index_catalogs.keys():
```

Comment:
This feels a bit strong. I can imagine someone having some canned dictionary of index catalogs, but not always using all of them. Could just check that `np.all(values.keys in index_catalog.keys())` or however that's written.